source/sas_compress.cpp

```diff
@@ -101,18 +171,16 @@ SAS::Compressor::Compressor()
 }
 
 /// Compressor destructor.  Terminates the zlib compressor.
-SAS::Compressor::~Compressor()
+ZlibCompressor::~ZlibCompressor()
 {
   deflateEnd(&_stream);
 }
 
 /// Compresses the specified string using the optional profile.
-std::string SAS::Compressor::compress(const std::string& s, const Profile* profile)
+std::string ZlibCompressor::compress(const std::string& s, std::string dictionary)
```

Comment:
Why this change?  Isn't it still reasonable to pass the full Profile in, rather than pulling out the dictionary?